packages/core/src/modules/discover-features/DiscoverFeaturesModule.ts

```diff
@@ -82,18 +86,34 @@ export class DiscoverFeaturesModule {
     return isProtocolSupported
   }
 
-  public async queryFeatures(connectionId: string, options: { query: string; comment?: string }) {
+  public async queryFeatures(connectionId: string, options: { query: string | FeatureQuery[]; comment?: string }) {
     const connection = await this.connectionService.getById(connectionId)
 
-    const queryMessage = await this.discoverFeaturesService.createQuery(options)
+    const queryMessage =
+      typeof options.query === 'string'
+        ? await this.discoverFeaturesService.createQuery({
+            query: options.query,
+            comment: options.comment,
+          })
+        : await this.discoverFeaturesV2Service.createQuery({ queries: options.query })
```

Comment:
```suggestion
        : await this.discoverFeaturesV2Service.createQuery({ query: options.query })
```

Or is there a reason for it being plural?

Reply:
Discover Features V1 has a "query" message that allows sending a single query, while V2 introduces "queries" allowing multiple queries. 

In this case I think it's more a copy-paste issue, as the method for `V2DiscoverFeaturesService` should be called `createQueries`.